src/modules/GameConstants.ts

```diff
@@ -1156,6 +1156,82 @@ export const TemporaryBattles = [
     'Ultra Megalopolis',
 ];
 
+export enum ShardTraderLocations {
```

Comment:
I don't like this way of connecting the trades to the city. It can somewhat easily go wrong, and our building-script don't help us.
Instead, we could create all the ShardMasterShops in ShardDeal.ts. We can still see them in the townlist (might need to add a reference).
Then we can give them the deals in the constructor, so we are sure they are connected.
Then you can get all the information you want from `ShopHandler.shopObservable().deal`.